relayer-cli/src/commands/create/channel.rs

```diff
@@ -1 +1,151 @@
+use ibc::Height;
+use ibc_relayer::channel::Channel;
+use ibc_relayer::config::StoreConfig;
+use ibc_relayer::connection::Connection;
+use ibc_relayer::foreign_client::ForeignClient;
+
+use crate::commands::cli_utils::{spawn_chain_runtime, ChainHandlePair, SpawnOptions};
+use crate::conclude::{exit_with_unrecoverable_error, Output};
+use crate::prelude::*;
+
+#[derive(Clone, Command, Debug, Options)]
+pub struct CreateChannelCommand {
+    #[options(
+        free,
+        required,
+        help = "Identifier of the side `a` chain for the new channel"
+    )]
+    chain_a_id: ChainId,
+
+    #[options(
+        free,
+        help = "Identifier of the side `b` chain for the new channel (optional)"
+    )]
+    chain_b_id: Option<ChainId>,
+
+    #[options(
+        help = "Identifier of the connection on chain `a` to use in creating the new channel"
+    )]
+    connection_a: Option<ConnectionId>,
+
+    #[options(
+        no_short,
+        required,
+        help = "Identifier of the side `a` port for the new channel"
+    )]
+    port_a: PortId,
+
+    #[options(
+        no_short,
+        required,
+        help = "Identifier of the side `b` port for the new channel"
+    )]
+    port_b: PortId,
+
+    #[options(help = "The order for parametrizing the new channel")]
+    order: Order,
+
+    #[options(help = "The version for parametrizing the new channel")]
+    version: String,
+}
+
+impl Runnable for CreateChannelCommand {
+    fn run(&self) {
+        match &self.chain_b_id {
+            None => self.run_reusing_connection(),
+            Some(chain_b) => self.run_using_new_connection(chain_b),
+        }
+    }
+}
+
+impl CreateChannelCommand {
+    // Creates a new channel, as well as a new underlying connection and clients.
+    fn run_using_new_connection(&self, chain_b_id: &ChainId) {
+        let config = app_config();
+
+        let spawn_options = SpawnOptions::override_store_config(StoreConfig::memory());
+
+        let chains =
+            ChainHandlePair::spawn_with(spawn_options, &config, &self.chain_a_id, chain_b_id)
+                .unwrap_or_else(exit_with_unrecoverable_error);
+        info!(
+            "Creating new clients, new connection, and a new channel with order {:?} and version {}",
+            self.order, self.version
+        );
+
+        let client_a = ForeignClient::new(chains.src.clone(), chains.dst.clone())
+            .unwrap_or_else(exit_with_unrecoverable_error);
+        let client_b = ForeignClient::new(chains.dst.clone(), chains.src.clone())
+            .unwrap_or_else(exit_with_unrecoverable_error);
+
+        // Create the connection.
+        // TODO: pass the `delay` parameter here.
+        let con =
+            Connection::new(client_a, client_b, 0).unwrap_or_else(exit_with_unrecoverable_error);
+
+        // Finally create the channel.
+        let chan_res = Channel::new(con, self.order, self.port_a.clone(), self.port_b.clone());
```

Comment:
As `Channel::new` returns a `Result`, we should check that it succeeded here.

Reply:
I think we should also rename `new` (which is typically used for constructors with no side effects) to eg. `handshake`.

Reply:
Agree. It's not clear to me what kind of output would be ideal here. I left this unfinished, and did similarly for Connection::new in a previous PR:

https://github.com/informalsystems/ibc-rs/blob/14684953690643a8e399e624bae7d383aa72888d/relayer-cli/src/commands/create/connection.rs#L95

The last decision regarding output was that we switch to human-friendly format by default (cf. ADR06). I suggest we address the ::new() and output problem here with an "output & logging" meta-issue, once we agree that human-friendly output is the way to go.

Reply:
Agreed that we can figure out a better story for logging at a later point, but I still think we should explicitly check whether the handshake has succeeded or not, much like you do for `Connection::find` a few lines above.

Reply:
Makes sense!

8137cbc2..9e666e91